hu/bme/annaATbarbies/sokoban/model/field/Trap.java

```diff
@@ -15,20 +15,73 @@
      */
     @Override
     public void accept(Pushable p) {
-
+    	SkeletonHelper.appendIndent();
+    	SkeletonHelper.write("Trap accept function.");
+    	SkeletonHelper.write("Is the trap active? 1: Yes; 2: No");
+    	int responseNum = SkeletonHelper.readInt();
+    	
+    	if(responseNum==1) {
+    		p.die();
+    	}
+    	else {
+    		new Trap().removePushable();
+        	this.setPushable(p);
+    	}
+    	
+    	SkeletonHelper.popIndent();
+    }
+    
+    /**
+     * Lekérdezi a csapda állapotát, ha nem aktív, akkor
+     * visszaadja azt a tolható elemet, ami a lépni kívánó elem előtt van, tehát amit el kell tolnia.
+     */
+    @Override
+    public pushable GetObstacle() {
+    	SkeletonHelper.appendIndent();
+    	SkeletonHelper.write("Trap getObstacle function.");
+    	SkeletonHelper.write("Is the trap active? 1: Yes; 2: No");
+    	int trapActive = SkeletonHelper.readInt();
+    	
+    	if(trapActive==2) {
+	    	SkeletonHelper.write("What should be on the field? 1: Empty; 2: Worker; 3: Box;");
+	    	int responseNum = SkeletonHelper.readInt();
+	    	
+	    	Pushable ret;
+	    	switch (responseNum) {
+	    		default:
+	    	    case 1: ret = null;
+	    	    	break;
+	    	    case 2: ret = new Worker();
+	    	    	break;
+	    	    case 3: ret = new Box();
+	    	    	break;
+	    	}
+	
+	    	SkeletonHelper.popIndent();
+	    	return ret;
+    	}
+    	else {
+    		SkeletonHelper.popIndent();
+    		return null;
+    	}
     }
 
     /**
      * kinyitja a csapdát, megöli aki rajta áll.
      */
     public void open() {
-
+    	SkeletonHelper.appendIndent();
+    	SkeletonHelper.write("Trap open function.");
+    	pushable.die();
```

Comment:
Nincs pushable változód.
Viszont szerintem elég annyi, hogy new Worker().die() úgyis a Pushable-ben van csak megírva, a Worker nem írja felül.
De az is lehet, hogy meg kellene kérdezni a felhasználót. Ki mit gondol?

Reply:
szerintem new Worker().die() lenne a jó